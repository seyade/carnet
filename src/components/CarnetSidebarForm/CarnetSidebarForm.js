import React, { Component } from 'react';
import axios from 'axios';

class CarnetSidebarForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      bookmarkTitle: '',
      url: '',
      info: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const target = e.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(e) {
    e.preventDefault();

    let payload = {
      bookmarkTitle: this.state.bookmarkTitle,
      url: this.state.url,
      info: this.state.info
    };

    this.postCarnet(payload);

    console.log('Payload: ', payload);
  }

  postCarnet(data) {
    return axios.post('http://localhost:3009/api/carnets', data)
      .then(response => {
        console.log('Response: ', response.message);
      });
  }

  render() {
    return (
      <div className="carnet-form-panel">
        <form className="carnet-form" onSubmit={this.handleSubmit}>
          <div className="form-group">
            <label htmlFor="carnet-bookmarktitle-field">Bookmark Title</label>
            <input className="form-control"
                   type="text"
                   placeholder="bookmark title..."
                   id="carnet-bookmarktitle-field"
                   name="bookmarkTitle"
                   value={this.state.bookmarkTitle}
                   onChange={this.handleChange} />
          </div>

          <div className="form-group">
            <label htmlFor="carnet-url-field">Url</label>
            <input className="form-control"
                   type="text"
                   placeholder="url..."
                   id="carnet-url-field"
                   name="url"
                   value={this.state.url}
                   onChange={this.handleChange} />
          </div>

          <div className="form-group">
            <label htmlFor="carnet-info-field">Info</label>
            <textarea className="form-control"
                      placeholder="info..."
                      id="carnet-info-field"
                      rows="5"
                      cols="20"
                      name="info"
                      value={this.state.info}
                      onChange={this.handleChange}>
            </textarea>
          </div>

          <input className="btn btn-primary" type="submit" value="Save" />
        </form>
        <div className="post-results-panel">
          <h2>{}</h2>
        </div>
      </div>
    );
  }
}

export default CarnetSidebarForm;
