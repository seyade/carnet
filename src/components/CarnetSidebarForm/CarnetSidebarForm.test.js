// jest.mock('../CarnetSidebarForm/CarnetSidebarForm'); // mock the component before importing it using the right path
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CarnetSidebarForm from '../CarnetSidebarForm/CarnetSidebarForm';

describe('CarnetSidebarForm', () => {
  let carnetform;

  beforeEach(() => {
    carnetform = mount(<CarnetSidebarForm />);
  });

  it('renders CarnetSidebarForm as intended', () => {
    const component = renderer.create(<CarnetSidebarForm />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('has field for bookmark title', () => {
    expect(carnetform.find('#carnet-bookmarktitle-field').length).toBe(1);

    carnetform.find('#carnet-bookmarktitle-field').node.value = 'Kenichi Strongest disciple';
    expect(carnetform.find('#carnet-bookmarktitle-field').node.value).toEqual('Kenichi Strongest disciple');
  });

  it('has field for url', () => {
    expect(carnetform.find('#carnet-url-field').length).toBe(1);

    carnetform.find('#carnet-url-field').node.value = 'http://www.kenichi.co';
    expect(carnetform.find('#carnet-url-field').node.value).toEqual('http://www.kenichi.co');
  });

  it('has field for info', () => {
    expect(carnetform.find('#carnet-info-field').length).toBe(1);

    carnetform.find('#carnet-info-field').node.value = 'Kenichi Strongest disciple. He trains at Ryuzanpaku school.';
    expect(carnetform.find('#carnet-info-field').node.value).toEqual('Kenichi Strongest disciple. He trains at Ryuzanpaku school.');
  });

  it('should submit form data', () => {
    carnetform.instance().handleSubmit = jest.fn();
    carnetform.update();
    carnetform.find('.carnet-submit-btn').simulate('submit');

    expect(carnetform.find('.carnet-submit-btn').first().length).toEqual(1);
    expect(carnetform.instance().handleSubmit).toHaveBeenCalled();
  });
});
