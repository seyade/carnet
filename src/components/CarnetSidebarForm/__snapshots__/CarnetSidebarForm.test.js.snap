exports[`CarnetSidebarForm renders CarnetSidebarForm as intended 1`] = `
<div
  className="carnet-form-panel">
  <form
    className="carnet-form"
    onSubmit={[Function]}>
    <div
      className="form-group">
      <label
        htmlFor="carnet-bookmarktitle-field">
        Bookmark Title
      </label>
      <input
        className="form-control"
        id="carnet-bookmarktitle-field"
        name="bookmarkTitle"
        onChange={[Function]}
        placeholder="bookmark title..."
        type="text"
        value="" />
    </div>
    <div
      className="form-group">
      <label
        htmlFor="carnet-url-field">
        Url
      </label>
      <input
        className="form-control"
        id="carnet-url-field"
        name="url"
        onChange={[Function]}
        placeholder="url..."
        type="text"
        value="" />
    </div>
    <div
      className="form-group">
      <label
        htmlFor="carnet-info-field">
        Info
      </label>
      <textarea
        className="form-control"
        cols="20"
        id="carnet-info-field"
        name="info"
        onChange={[Function]}
        placeholder="info..."
        rows="5"
        value="" />
    </div>
    <input
      className="carnet-submit-btn btn btn-primary"
      type="submit"
      value="Save" />
  </form>
  <div
    className="post-results-panel">
    <h2 />
  </div>
</div>
`;
